File: DS_P1_S3/C++/CadenaFiltros.cpp
#include "CadenaFiltros.h"

void CadenaFiltros::addFiltro(Filtro* filtro) {
    filtros.push_back(filtro);
}

FormularioPrestamo* CadenaFiltros::ejecutar(FormularioPrestamo* f) {
    for( int i=0; i < filtros.capacity(); i++){
        f = filtros.at(i)->ejecutar(f);
    }

    return f;
}

Prestamo CadenaFiltros::pedirPrestamo(FormularioPrestamo* f) {
    setPrestamo(this->prestamo.pedirPrestamo(f->getCuantia(),f->getCuotas(),f->getInteres()));
    return this->prestamo;
}

CadenaFiltros::CadenaFiltros() {
}

void CadenaFiltros::setPrestamo(Prestamo p) {
    this->prestamo = p;
}

File: DS_P1_S3/Dart/p1_s3/lib/CadenaFiltros.dart
import 'FormularioPrestamo.dart';
import 'prestamo.dart';
import 'Filtro.dart';

class CadenaFiltros{
  Prestamo prestamo;
  List<Filtro> filtros = List<Filtro>();

  CadenaFiltros(){
    prestamo = new Prestamo.otroConstructor();
  }

  void addFiltro(Filtro f){
    filtros.add(f);
  }
  FormularioPrestamo ejecutar(FormularioPrestamo f){
    for( var e in filtros){
      f = e.ejecutar(f);
    }
    return f;
  }
  void setPrestamo(Prestamo p){
    prestamo = p;
  }
  Prestamo pedirPrestamo(FormularioPrestamo f){
    setPrestamo(prestamo.pedirPrestamo(f));
    return prestamo;
  }

}